app/controllers/UploadController.scala

```diff
@@ -127,4 +141,9 @@ class UploadController @Inject()(
       <Resource>NoFileReference</Resource>
       <RequestId>SomeRequestId</RequestId>
     </Error>""")
+
+  def generateChecksum(fileBytes: Array[Byte]): String = {
+    val checksum = MessageDigest.getInstance("SHA-1").digest(fileBytes)
```

Comment:
SHA256 instead of sha? I think it's important that stub returns exactly the same results here as real service.